mixpanel.go

```diff
@@ -183,29 +219,38 @@ func (m *mixpanel) send(eventType string, params interface{}, autoGeolocate bool
 		return wrapErr(bodyErr)
 	}
 
-	if strBody := string(body); strBody != "1" && strBody != "1\n" {
-		return wrapErr(&ErrTrackFailed{Body: strBody, Resp: resp})
+	type verboseResponse struct {
+		Error  string `json:"error"`
+		Status int    `json:"status"`
+	}
+
+	var jsonBody verboseResponse
+	json.Unmarshal(body, &jsonBody)
+
+	if jsonBody.Status != 1 {
+		return wrapErr(&ErrTrackFailed{Message: jsonBody.Error})
 	}
 
 	return nil
 }
 
 // New returns the client instance. If apiURL is blank, the default will be used
 // ("https://api.mixpanel.com").
-func New(token, apiURL string) Mixpanel {
-	return NewFromClient(http.DefaultClient, token, apiURL)
+func New(token, secret, apiURL string) Mixpanel {
+	return NewFromClient(http.DefaultClient, token, secret, apiURL)
 }
 
 // Creates a client instance using the specified client instance. This is useful
 // when using a proxy.
-func NewFromClient(c *http.Client, token, apiURL string) Mixpanel {
+func NewFromClient(c *http.Client, token, secret, apiURL string) Mixpanel {
```

Comment:
This is a breaking change. It shouldn't affect anything we are using because of the fork but is there any way of having another constructor that calls the original NewFromClient and adds the secret too? `NewFromClientWithSecret` which calls `NewFromClient`?

Reply:
Yeah good idea. This won't get un-forked unless we do this I think.